### ISSUE TITLE
Random number generation yields different values on different devices, despite the same manual seed.

### ISSUE THREAD

**Issue by chausies** [pytorch/pytorch#79496]
### 🐛 Describe the bug

Given the **same** manual seed, cpu and cuda generate two different sets of random numbers. See the following code.
```python3
import torch
cuda = torch.device("cuda")
torch.manual_seed(10)
torch.randn(3)
# always yields tensor([-0.6014, -1.0122, -0.3023])
torch.manual_seed(10)
torch.randn(3, device=cuda)
# always yields tensor([-0.1029,  1.6810, -0.2708], device='cuda:0')
```
But that shouldn't be the case, as it breaks reproducibility. At the beginning of pytorch programs, you often select either the CPU to use as the device, or the GPU if its available. But now, depending on which device is selected, different sets of random numbers will be produced from a program that was otherwise seeded to run a single deterministic way.

### Versions

Collecting environment information...
PyTorch version: 1.11.0+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Mar 15 2022, 12:22:08)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Ti
Nvidia driver version: 512.95
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.20.2
[pip3] torch==1.11.0+cu113
[pip3] torchaudio==0.11.0+cu113
[pip3] torchvision==0.12.0+cu113
[conda] Could not collect

cc @pbelevich

**Comment by ngimel** [comment]
This is expected https://pytorch.org/docs/stable/notes/randomness.html
